src/nvim/terminal.c

```diff
@@ -380,9 +380,11 @@ void terminal_enter(void)
   int save_w_p_cul = curwin->w_p_cul;
   int save_w_p_cuc = curwin->w_p_cuc;
   int save_w_p_rnu = curwin->w_p_rnu;
+  int save_w_p_list = curwin->w_p_list;
   curwin->w_p_cul = false;
   curwin->w_p_cuc = false;
   curwin->w_p_rnu = false;
+  curwin->w_p_list= false;
```

Comment:
We don't want to do this on terminal_enter because then the user's preference will get reverted each time they `i` into the terminal. (Also the "They are nonsense" comment on line 377 is a non-sequitur for this option.)

Instead let's do it in terminal_open (line 240), which still gives the user a chance to force the option in e.g. `TermOpen` autocmd.